### PR TITLE
Pipeline: support partition table scan without optimize

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #6518 

Problem Summary:

### What is changed and how it works?
- rename `force_enable_pipeline` to `enforce_enable_pipeline`.
- support partition table scan without optimize ref #4474

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp [hunk 1]
```diff
@@ -1064,13 +1064,17 @@ SourceOps DAGStorageInterpreter::buildLocalSourceOps(
...
     SourceOps source_ops;
     for (const auto & table_query_info : table_query_infos)
     {
         const TableID table_id = table_query_info.first;
         const SelectQueryInfo & query_info = table_query_info.second;
-        source_ops = buildLocalSourceOpsForPhysicalTable(exec_status, table_id, query_info, max_block_size);
+
+        auto table_source_ops = buildLocalSourceOpsForPhysicalTable(exec_status, table_id, query_info, max_block_size);
+        for (auto & source : table_source_ops)
+            source_ops.push_back(std::move(std::move(source)));
```
Comment (Lloyd-Pottiger) [inline dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp:1077]: why `std::move(std::move(source))`?

Comment (SeaRise) [reply dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp]: My mistake...fix

File: dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp [hunk 2]
```diff
@@ -1064,13 +1064,17 @@ SourceOps DAGStorageInterpreter::buildLocalSourceOps(
...
     SourceOps source_ops;
     for (const auto & table_query_info : table_query_infos)
     {
         const TableID table_id = table_query_info.first;
         const SelectQueryInfo & query_info = table_query_info.second;
-        source_ops = buildLocalSourceOpsForPhysicalTable(exec_status, table_id, query_info, max_block_size);
+
+        auto table_source_ops = buildLocalSourceOpsForPhysicalTable(exec_status, table_id, query_info, max_block_size);
+        for (auto & source : table_source_ops)
+            source_ops.push_back(std::move(source));
```
Comment (Lloyd-Pottiger) [inline dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp:1077]: use `insert` instead?

File: dbms/src/Flash/Pipeline/Pipeline.cpp [hunk 3]
```diff
@@ -26,6 +26,8 @@
 #include <Interpreters/Settings.h>
 #include <tipb/select.pb.h>
 
+#include <magic_enum.hpp>
+
```
Comment (Lloyd-Pottiger) [inline dbms/src/Flash/Pipeline/Pipeline.cpp:30]: do we need this header?

Comment (SeaRise) [reply dbms/src/Flash/Pipeline/Pipeline.cpp:30]: To use `magic_enum::enum_name` in https://github.com/pingcap/tiflash/pull/7474/files#diff-dce9669ca981e1c7ff8dd7f7f327469a65f85c31bc7fc9aa15a4dd6a9a46ea58R281-R283.

File: dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp (hunk 2 above)
Comment (SeaRise) [reply dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp]: ok, updated to `source_ops.insert(source_ops.end(), std::make_move_iterator(table_source_ops.begin()), std::make_move_iterator(table_source_ops.end()));`

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**